### PR TITLE
Remove section on `git revert`

### LINKED ISSUE #24: Add specific use cases for rewriting history
> I was thinking of just one-line descriptions of the sorts of scenarios these techniques can help with, e.g. just add a paragraph like so:

```
For example:
- You have already made a commit, but realise there are other changes you forgot to include
- You have made a commit, but have changed your mind and want to remove this change from your history
- You want to undo a previous commit made by yourself or a colleague (and make a record of this in the history)
- You want to "move" one or more commits so they are based on top of some other work (e.g. new changes made to the main branch
```

_Originally posted by @alexdewar in https://github.com/ImperialCollegeLondon/intermediate_grad_school_git_course/pull/19#discussion_r1043270036_

### LINKED ISSUE #91: Remove section on `git revert`
We already teach students about `git revert` in the intro course anyway and we're trying to free up some time for the final exercise, so we might as well drop it. (It doesn't fit terribly well in a section on rewriting history anyway.)

### PR DESCRIPTION
**Pull request by alexdewar**
I've removed the section on `git revert`, as we discussed. I also fixed up some code blocks and added some text explaining why you might need to rewrite your history (#24).

Closes #91. Closes #24.

### CODE REVIEW

File: _episodes/l1-04-rewriting_history.md [hunk 1]
```diff
@@ -28,8 +27,99 @@ time, it also lets you to modify the timeline of commits. There are several tota
...
+for your benefit and that of your collaborators. But, first, we will look at another
+useful command.
+
+### Set aside your work safely with `stash`
+
+It is not rare that, while you are working on some feature, you need to check something
+else in another branch. Very often this is the case when you want to try some
+contributor's code as part of a pull request review process (see next episodes). You
+can commit the work you are doing, but if it is not in a state ready to be committed,
+what would you do?
```
Comment (SaranjeetKaur) [inline _episodes/l1-04-rewriting_history.md:50]:
```suggestion
It is not rare that, while you are working on some feature, you need to check something
else in another branch. Very often this is the case when you want to try some
contributor's code as part of a pull request review process (see next episodes). You
can commit the work you are doing, but if it is not in a state ready to be committed,
what would you do? Or you start working on a branch only to realise that it is not the one that you were planning to work on?
```

File: _episodes/l1-04-rewriting_history.md [hunk 2]
```diff
@@ -39,32 +129,37 @@ made, maybe adding some files you forgot to stage or fixing a typo in the commit
...
 
-The next level of complexity rewriting history is `reset`: it lets you redo the
-last (or last few) commit(s) you made so you can incorporate more changes, fix an error
-you have spotted and that is worth incorporating as part of that commit and not as a
-separate one or just improve your commit message.
+The next level of complexity rewriting history is `reset`: it lets you redo the last (or
+last few) commit(s) you made so you can incorporate more changes, fix an error you have
+spotted and that is worth incorporating as part of that commit and not as a separate one
+or just improve your commit message. Unlike `git revert`, `git reset` will
+retrospectively alter your commit history, so should **not** be used where you have
```
Comment (SaranjeetKaur) [inline _episodes/l1-04-rewriting_history.md:156]:
```suggestion
retrospectively alter your commit history, so it should **not** be used when you have
```

File: _episodes/l1-04-rewriting_history.md [hunk 3]
```diff
@@ -81,17 +176,35 @@ accordingly. In other words, **any work done since the chosen commit will be com
...
+> ## `reset` vs `revert`
+>
+> `git revert` was [discussed in the introductory course](https://imperialcollegelondon.github.io/introductory_grad_school_git_course/l1-02-committing-history/index.html).
+>
+> `reset` and `revert` both let you undo things done in the past, but they both have
+> very different use cases.
+>
+> - `reset` uses brute force, potentially with destructive consequences, to
+>   make those changes and is suitable only if the work has not been shared with others
+>   already. Use when you want to get rid of recent work you're not happy with and start
```
Comment (SaranjeetKaur) [inline _episodes/l1-04-rewriting_history.md:201]:
```suggestion
>   already. Use it when you want to get rid of recent work you're not happy with and start
```

File: _episodes/l1-04-rewriting_history.md [hunk 4]
```diff
@@ -81,17 +176,35 @@ accordingly. In other words, **any work done since the chosen commit will be com
...
+>   all over again.
+> - `revert` is more lightweight and surgical, to target specific changes and creating
+>   new commits to history. Use when code has already been shared with others or when
```
Comment (SaranjeetKaur) [inline _episodes/l1-04-rewriting_history.md:204]:
```suggestion
>   new commits to history. Use it when code has already been shared with others or when
```

**Review by SaranjeetKaur — APPROVED** [review]
Thanks @alexdewar! This looks nice. I have added only some minor comments.

**Review by AdrianDAlessandro — COMMENTED** [review]
I think this is good. I haven't built it locally, but we'll do a review at a later time.

Are we still planning to move the `git stash` stuff to the Merge Conflicts section? I think it fits better there anyway.

**Comment by alexdewar** [comment]
@AdrianDAlessandro We can do. I've made a separate issue for it: #94

**Pull request merged**